packages/connect-webextension/src/index.ts

```diff
@@ -96,7 +93,13 @@ const init = (settings: Partial<ConnectSettings> = {}): Promise<void> => {
                 // in this case, settings will be validated in popup
                 payload: { settings: _settings },
             });
-            handshakePromise.resolve();
+            _popupManager.handshakePromise.resolve();
+        }
+        if (message.type === 'popup-closed') {
```

Comment:
there is POPUP.CLOSED constant. I'd use it here.

Reply:
Yes, updated.